haystack/nodes/preprocessor/preprocessor.py

```diff
@@ -375,13 +389,20 @@ def split(
         text = document.content
         headlines = document.meta["headlines"] if "headlines" in document.meta else []
 
-        if split_respect_sentence_boundary and split_by == "word":
-            text_splits, splits_pages, splits_start_idxs = self._split_by_word_respecting_sent_boundary(
-                text=text, split_length=split_length, split_overlap=split_overlap
+        if split_respect_sentence_boundary and split_by in ["word", "token"]:
+
+            def split_function(text):
+                if split_by == "token":
+                    return self._split_tokens(text, tokenizer=tokenizer)
+                else:
+                    return text.split()
+
+            text_splits, splits_pages, splits_start_idxs = self._split_into_units_respecting_sent_boundary(
+                text=text, split_length=split_length, split_overlap=split_overlap, split_function=split_function
```

Comment:
Why passing an inner function instead of simply `split_by` and the `tokenizer`? There seem to be no need for this convolution.

Reply:
The reason is that `split()` is called in multiple places and also in another function called by `_split_into_units_respecting_sent_boundary` in the function, so it makes more sense to me to only check split_by once and then reuse the split function. Also, one extra parameter is better than two.
But I will of course change it, if you prefer.